microservices/download/utils.js

```diff
@@ -38,6 +38,10 @@ function query_os(req) {
   let raw = req; // The URL string containing any number of query params.
   let prov = undefined;
 
+  if (typeof raw !== "string") {
+    return false;
+  }
+
```

Comment:
I am unsure if `raw` can possibly not be a `string`, since it starts with e.g. `https://...` and then the user-provided query parameters afterward.

That said, being conservative around that possibility feels more worthwhile than YOLOing if there is any remaining doubt -- And maybe there's a little doubt, so ultimately 👍 .

Reply:
This change was actually in a reaction to us calling `split()` later on the raw value and crashing the microservice because the method wasn't available.

If I had to assume, the fact that it can be something not a string depends on how the HTTP handler built into NodeJS decodes this data, since I'm pretty sure we are accessing the parameters, rather than the full raw URL string

Reply:
Ah, okay.

I think I got a bit confused by var names and code comments nearby.

Maybe I can think of a way to clarify the argument names, such as `req` here, and the `path` var name in the call site: https://github.com/pulsar-edit/package-frontend/blob/6bdf643c6aa237e1969cd2675a7dfc02884d64cf/microservices/download/index.js#L12

As well as making this comment a bit less ambiguous, since I mistakenly read it as implying the argument would take the form `https://...?param=val&param2=val2...`:
```js
let raw = req; // The URL string containing any number of query params.
```

Reply:
@DeeDeeG So just double checked, and what you thought does have a lot of merit.

In `index.js` we take the path via `req.url` and split this on `?`. Meaning we did at one point have the full path, as you envisioned it. But then we just pass the second part of that array to these query parameter handlers. Which, in the event there was nothing to actually split, and the resulting array has a single item, then we would be passing `null` meaning it's not a string.

So absolutely some better comments here and variables names aren't a bad idea.